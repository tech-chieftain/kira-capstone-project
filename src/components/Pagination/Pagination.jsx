import React from "react";
import ReactPaginate from "react-paginate";
import PropTypes from "prop-types";

const Pagination = ({ pageCount, handlePageClick, ariaLabel }) => (
  <nav aria-label={ariaLabel}>
    <ReactPaginate
      previousLabel="Previous"
      nextLabel="Next"
      breakLabel="..."
      pageCount={pageCount}
      marginPagesDisplayed="1"
      pageRangeDisplayed="2"
      containerClassName="pagination justify-content-center"
      pageClassName="page-item"
      pageLinkClassName="page-link"
      previousClassName="page-item"
      nextClassName="page-item"
      previousLinkClassName="page-link"
      nextLinkClassName="page-link"
      breakClassName="page-item"
      breakLinkClassName="page-link"
      activeClassName="active"
      onPageChange={(data) => handlePageClick(data.selected)}
    />
  </nav>
);

Pagination.propTypes = {
  pageCount: PropTypes.number.isRequired,
  handlePageClick: PropTypes.func.isRequired,
  ariaLabel: PropTypes.string,
};

Pagination.defaults = {
  ariaLabel: "Page navigation",
};

export default Pagination;
