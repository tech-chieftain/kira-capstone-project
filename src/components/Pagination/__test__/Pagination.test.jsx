import * as React from "react";
import * as renderer from "react-test-renderer";
import Pagination from "../Pagination";

it("renders Pagination", () => {
  const tree = renderer
    .create(
      <Pagination pageCount={10} handlePageClick={(pageNum) => pageNum} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
