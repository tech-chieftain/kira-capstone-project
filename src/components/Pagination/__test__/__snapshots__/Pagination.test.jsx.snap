// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Pagination 1`] = `
<nav>
  <ul
    className="pagination justify-content-center"
  >
    <li
      className="page-item disabled"
    >
      <a
        aria-disabled="true"
        aria-label="Previous page"
        className="page-link"
        onClick={[Function]}
        onKeyPress={[Function]}
        rel="prev"
        role="button"
        tabIndex="0"
      >
        Previous
      </a>
    </li>
    <li
      className="page-item active"
    >
      <a
        aria-current="page"
        aria-label="Page 1 is your current page"
        className="page-link"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        1
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        aria-current={null}
        aria-label="Page 2"
        className="page-link"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        2
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        aria-current={null}
        aria-label="Page 3"
        className="page-link"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        3
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        ...
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        aria-current={null}
        aria-label="Page 10"
        className="page-link"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        10
      </a>
    </li>
    <li
      className="page-item"
    >
      <a
        aria-disabled="false"
        aria-label="Next page"
        className="page-link"
        onClick={[Function]}
        onKeyPress={[Function]}
        rel="next"
        role="button"
        tabIndex="0"
      >
        Next
      </a>
    </li>
  </ul>
</nav>
`;
